engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -272,12 +273,22 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
    */
   private def generateCollectorOutput(collector: CollectorKey, shards: Iterable[BackendCallKey]): Try[CallOutputs] = Try {
     val shardsOutputs = shards.toSeq sortBy { _.index.fromIndex } map { e =>
-      fetchCallOutputEntries(e) map { _.outputs } getOrElse(throw new RuntimeException(s"Could not retrieve output for shard ${e.scope} #${e.index}"))
+      fetchCallOutputs(e) getOrElse(throw new RuntimeException(s"Could not retrieve output for shard ${e.scope} #${e.index}"))
     }
     collector.scope.task.outputs map { taskOutput =>
-      val wdlValues = shardsOutputs.map(s => s.getOrElse(taskOutput.name, throw new RuntimeException(s"Could not retrieve output ${taskOutput.name}")))
-      val arrayOfValues = new WdlArray(WdlArrayType(taskOutput.wdlType), wdlValues)
-      taskOutput.name -> CallOutput(arrayOfValues, workflow.hash(arrayOfValues))
+      val callOutputs = shardsOutputs.map(s => s.getOrElse(taskOutput.name, throw new RuntimeException(s"Could not retrieve output ${taskOutput.name}")))
+      val arrayOfValues = new WdlArray(WdlArrayType(taskOutput.wdlType), callOutputs map { _.wdlValue })
+      val hashes = callOutputs map { output => {
+        output.hash orElse workflow.hash(output.wdlValue)
+      } }
+      val hash = if (hashes forall { _.isDefined }) {
+        val concatenatedHashes = hashes map { _.get.value } mkString ""
+        val hash = SymbolHash((WdlArray.getClass.getCanonicalName + concatenatedHashes).md5Sum)
```

Comment:
This is very dirty, it emulates the aggregation logic in wdl4s, which is private and can't be re-used. A cleaner way would be to patch wdl4s 0.4 to factor out this logic and update the dependency here in 0.19_hotfix.